### DISCUSSION TITLE
Feature requests for standard financial report

### DISCUSSION THREAD

**Discussion by sboele** [firefly-iii/firefly-iii#4320] [Firefly III]
I have a couple of feature requests for the standard financial report

# Grouping
When you create a standard financial report over multiple years it groups data per year. Even if the report starts halfway the year. It would be nice if we could select the grouping period (year, month). Right now it's hard to compare over multiple years.

Grouping per month works if you select a period in the same year for example: `/reports/default/allAssetAccounts/20200101/20201231`

![image](https://user-images.githubusercontent.com/1193437/105493438-b167b100-5cb9-11eb-89cd-fbed7fdcb6a5.png)

But if you select a period that spans multiple years for example: `/reports/default/allAssetAccounts/20190101/20210131` it will group everything per year.

![image](https://user-images.githubusercontent.com/1193437/105493395-a3199500-5cb9-11eb-9183-e08ebe1b79f2.png)

# Average column
For category table it would be nice that next to the sum column an average column would be added. Other ideas I had was to add icons indicating that a category in a certain period is higher/lower than the average.

**Comment by JC5** [comment]
Thanks for the suggestions. I'll remember them for the new layout, which I'm currently building. Keep in mind that at no point Firefly III will give you options like "group per month or per year". [Sensible defaults](https://en.wikipedia.org/wiki/Convention_over_configuration) all the way. 

But I like your ideas and I've added them to my list!
